db/migrate/20190329201346_move_partner_status_to_an_interger_type.rb

```diff
@@ -0,0 +1,25 @@
+class MovePartnerStatusToAnIntergerType < ActiveRecord::Migration[5.2]
+  def up
+    # 1. add_column that is an integer type where we can move our string to ints
+    add_column :partners, :temp_status, :integer, default: 0
+    # "Pending" => 0; etc.
+    # 2. Map status data from the "status" to our new "temp_status"
+    puts "updating Pending partners. will update #{Partner.where(status: "Pending").count} partners"
+    Partner.where(status: "Pending").update_all(temp_status: 0)
+    puts "updating Awaiting Review partners. will update #{Partner.where(status: "Awaiting Review").count} partners"
+    Partner.where(status: "Awaiting Review").update_all(temp_status: 1)
+    puts "updating Approved partners. will update #{Partner.where(status: "Approved").count} partners"
+    Partner.where(status: "Approved").update_all(temp_status: 2)
+    # 3. Delete "status"
+    puts "removing status column"
+    remove_column :partners, :status
+    # 4. Rename "temp_status" -> "status"
+    puts "renaming temp_status to status"
+    rename_column :partners, :temp_status, :status
+  end
```

Comment:
@armahillo Here, we're we've worked out a simple data migration for migrations for `status` going from string to integer for support of status enum. 

Part of what we've done here introduces a default value `0` for `pending`. Given enums for status, having a default value makes sense to me. 

We need to work out if this test is still relevant as the _status is not longer changing_. https://github.com/Nirvikalpa108/diaper/blob/742-alert-partner-status/spec/jobs/update_diaper_partner_job_spec.rb#L17

What are your thoughts?

Reply:
If the default status is set to `pending`, then you could probably tweak the test to assert that it's getting set to pending by default (if another test already satisfies that, you can delete the "change" test)

Nice work, both of you!